activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-rest/src/test/java/org/activiti/cloud/services/modeling/rest/controller/GenericJsonModelTypeModelUpdateListenerControllerIT.java

```diff
@@ -0,0 +1,119 @@
+ * limitations under the License.
+ */
+package org.activiti.cloud.services.modeling.rest.controller;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.annotation.DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.webAppContextSetup;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.activiti.cloud.modeling.api.JsonModelType;
+import org.activiti.cloud.modeling.api.Model;
+import org.activiti.cloud.modeling.api.ModelUpdateListener;
+import org.activiti.cloud.modeling.repository.ModelRepository;
+import org.activiti.cloud.services.modeling.config.ModelingRestApplication;
+import org.activiti.cloud.services.modeling.entity.ModelEntity;
+import org.activiti.cloud.services.modeling.security.WithMockModelerUser;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.web.context.WebApplicationContext;
+
+/**
+ * Integration tests for models rest api dealing with Json models
+ */
+@ActiveProfiles(profiles = {"test", "generic"})
+@SpringBootTest(classes = ModelingRestApplication.class)
+@WebAppConfiguration
+@DirtiesContext(classMode = AFTER_EACH_TEST_METHOD)
+@WithMockModelerUser
+public class GenericJsonModelTypeModelUpdateListenerControllerIT {
+
+    @Autowired
+    private WebApplicationContext context;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private ModelRepository modelRepository;
+
+    @Autowired
+    private JsonModelType genericJsonModelType;
+
+    @SpyBean(name = "genericJsonModelUpdateListener")
+    private ModelUpdateListener genericJsonModelUpdateListener;
+
+    @SpyBean(name = "genericNonJsonModelUpdateListener")
+    private ModelUpdateListener genericNonJsonModelUpdateListener;
+
+    private MockMvc mockMvc;
+
+    private static final String GENERIC_MODEL_NAME = "simple-model";
+
+    @BeforeEach
+    public void setUp() {
+        mockMvc = webAppContextSetup(context).build();
+    }
+
+    @Test
+    public void should_callJsonModelUpdateListener_when_updatingModelContent() throws Exception {
```

Comment:
I'd merge the two tests of this class in a single one and merge the `verify` part (check that the listener for which the type matching is called and the one for which the type is not matching is not called). The name could be changed to something like `should_callUpdateListenerMatchingWithModelType_when_updatingModelContent`.

Reply:
OK, you are right, both tests do the same call to the API so we can do both verifications in the same test